### PR TITLE
prov/verbs: Remove deprecated FI_EP_RDM type

### LINKED ISSUE #4278: How about to remove verbs/RDM?
What we have:
1. libfabric uses RxM/verbs/MSG as a default
2. RxM/verbs pass all MPICH tests with MPICH/IMPI. And as far as I know, well tested with OpenMPI.
verbs/RDM has too much bugs and not stable.
3. RxM/verbs (latest one form master branch) shows better performance numbers and scalability

@shefty @a-ilango what do you think? If you agree, I'd remove verbs/RDM's code from libfabric in 1.7 release.

### PR DESCRIPTION
**Pull request by dmitrygladkov**
This patch prepares master branch for 1.7 release where `FI_EP_RDM` will be deleted.

Now Verbs provider supports only `FI_EP_DGRAM` and `FI_EP_MSG`.

The next step is to make possibility to use different EP types (`FI_EP_DGRAM`and `FI_EP_MSG`) simultaneously per single domain.

Fixes #4278 

Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

File: include/rdma/fabric.h [hunk 1]
```diff
@@ -287,8 +287,6 @@ enum {
 	 *  We will keep  this value in order to save binary compatibility.
 	 */
 	FI_PROTO_MXM,
-	FI_PROTO_IWARP_RDM,
-	FI_PROTO_IB_RDM,
```
Comment (shefty) [inline include/rdma/fabric.h]: These values must remain, or it breaks source and binary compatibility.  This modifies the values of enums defined further down.

File: src/fi_tostr.c [hunk 2]
```diff
@@ -248,8 +248,6 @@ static void fi_tostr_protocol(char *buf, uint32_t protocol)
 	CASEENUMSTR(FI_PROTO_PSMX2);
 	CASEENUMSTR(FI_PROTO_UDP);
 	CASEENUMSTR(FI_PROTO_SOCK_TCP);
-	CASEENUMSTR(FI_PROTO_IB_RDM);
-	CASEENUMSTR(FI_PROTO_IWARP_RDM);
```
Comment (shefty) [inline src/fi_tostr.c]: please keep this as well

File: include/rdma/fabric.h (hunk 1 above)
Comment (dmitrygladkov) [reply include/rdma/fabric.h]: Done

File: src/fi_tostr.c (hunk 2 above)
Comment (dmitrygladkov) [reply src/fi_tostr.c]: Done

**Pull request merged**